uniflow-test/src/test/kotlin/io/uniflow/test/StackFlowTest.kt

```diff
@@ -36,33 +36,34 @@ class StackFlowTest {
 
     @Test
     fun `empty state`() {
-        assertEquals(UIState.Empty, dataFlow.states.first())
+        dataFlow.hasStates(UIState.Empty)
+        dataFlow.hasNoEvent()
     }
 
     @Test
     fun `get all`() {
         dataFlow.getAll()
-        assertEquals(UIState.Empty, dataFlow.states[0])
-        assertEquals(TodoListState(emptyList()), dataFlow.states[1])
+        dataFlow.hasStates(UIState.Empty, TodoListState(emptyList()))
+        dataFlow.hasNoEvent()
     }
 
     @Test
     fun `add one`() {
         dataFlow.getAll()
         dataFlow.add("first")
-
-        assertEquals(UIState.Empty, dataFlow.states[0])
-        assertEquals(TodoListState(emptyList()), dataFlow.states[1])
-        assertEquals(TodoListState(listOf(Todo("first"))), dataFlow.states[2])
+        dataFlow.hasStates(
+                UIState.Empty,
+                TodoListState(emptyList()),
+                TodoListState(listOf(Todo("first"))))
+        dataFlow.hasNoEvent()
     }
 
     @Test
     fun `add one - fail`() {
         dataFlow.add("first")
 
-        assertEquals(UIState.Empty, dataFlow.states[0])
-
-        assertTrue(dataFlow.events[0] is UIEvent.BadOrWrongState)
+        dataFlow.hasStates(UIState.Empty)
+        dataFlow.hasEvents(UIEvent.BadOrWrongState(UIState.Empty))
```

Comment:
This is an example where my previous comment about checking order of both states and events applies:

This test should not only assert that the state is `Empty` and the event is `BadOrWrongState`, but also that they were observed in the correct order, because

1. `Empty`
1. `BadOrWrongState`

does not equal

1. `BadOrWrongState`
1. `Empty`

I think this is out of this PR's scope, though, so let's discuss this elsewhere.

Reply:
exact!